### DISCUSSION TITLE
Support of Platform-specific extensions

### DISCUSSION THREAD

**Discussion by JamesHemery** [expo/router#490] [Q&A]
Hi, 

I think that [platform specific extensions](https://reactnative.dev/docs/platform-specific-code#platform-specific-extensions) are not supported in the app folder, is this a problem on my side or is it known? As soon as I exit the app folder it works.

Thanks,

**Comment by wcastand** [comment]
platform specific extension are only a problem in the app folder indeed. if you use it later in one of your component it will work as expected. You just can't use platform-specific extensions for the routes.

**Reply by JamesHemery** [reply]
Yes indeed, one solution is to export a component that is outside the app folder into the route file. But the support is possible I think, maybe in V2.

**Reply by jmeijon** [reply]
I'm having an issue, I have a screen with with the name: screen.web.tsx and other screen.tsx and when I run on web the screen.tsx is called. I think that was supported on expo router. Also tried with screen.native.tsx (instead of having screen.tsx in opposite way) and didn't work. It's always calling the screen.tsx for all the platforms

**Reply by Shaswat-dev** [reply]
Did  you find a solution to this ?

**Comment by marklawlor** [comment]
We are experimenting with platform specific extensions for routes and `_layouts`. There no ETA or guarantee we will go ahead with this feature. You can track the initial progress here: https://github.com/expo/expo/pull/27408

Until then you can follow the documentation on how to create platform specific routes https://docs.expo.dev/router/advanced/platform-specific-modules/
